### PR TITLE
Support HTTPS readiness probe

### PR DESCRIPTION
**Pull request by clyang82**
fixed: https://github.com/efficientgo/e2e/issues/9

Signed-off-by: clyang82 <chuyang@redhat.com>

### CODE REVIEW

**Review by GiedriusS — APPROVED** [review]

File: env.go [hunk 1]
```diff
@@ -207,19 +208,24 @@ type ReadinessProbe interface {
 	Ready(runnable Runnable) (err error)
 }
 
-// HTTPReadinessProbe checks readiness by making HTTP call and checking for expected HTTP status code.
+// HTTPReadinessProbe checks readiness by making HTTP or HTTPS call and checking for expected HTTP/HTTPS status code.
 type HTTPReadinessProbe struct {
 	portName                 string
 	path                     string
+	scheme                   string
 	expectedStatusRangeStart int
```
Comment (matej-g) [inline env.go:216]: How about making this a bool? For now I cannot imagine scenario where we would need to specify more schemas, other than choosing between either HTTP or HTTPS.

**Review by matej-g — COMMENTED** [review]
Thanks for picking up this! I have one suggestion but overall looks good👍

File: env.go (hunk 1 above)
Comment (clyang82) [reply env.go:216]: I think it is better to align with kubernetes - https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#http-probes

Comment (matej-g) [reply env.go:216]: That's a fair point, agreed then :+1: Would be still nicer to have this as a typed string with either `HTTP` or `HTTPS` option, but that's more of a nit.

**Review by matej-g — APPROVED** [review]
Looks good, thanks!

cc @bwplotka could we get a merge on this if you're fine with the PR?

**Comment by clyang82** [comment]
@bwplotka PTAL. Thanks.

File: env.go [hunk 2]
```diff
@@ -207,19 +208,24 @@ type ReadinessProbe interface {
...
 	expectedStatusRangeEnd   int
 	expectedContent          []string
 }
 
-func NewHTTPReadinessProbe(portName string, path string, expectedStatusRangeStart, expectedStatusRangeEnd int, expectedContent ...string) *HTTPReadinessProbe {
+func NewHTTPReadinessProbe(portName, path, scheme string, expectedStatusRangeStart, expectedStatusRangeEnd int, expectedContent ...string) *HTTPReadinessProbe {
```
Comment (bwplotka) [inline env.go:221]: I would prefer having `NewHTTPReadinessProbe` and `NewHTTPSReadinessProbe` so we don't need this extra `""` everywhere. WDYT?

we could have `newHTTPReadinessProve(portName, path, scheme. ..` we should share.

**Review by bwplotka — APPROVED** [review]
Great, thank you!

One idea to improve the experience.

File: env.go (hunk 2 above)
Comment (clyang82) [reply env.go:221]: I do not like the extra ""  everywhere. And the method should be backward-compatible. so I agree to have `NewHTTPReadinessProbe` and `NewHTTPSReadinessProbe`.

What do you mean `newHTTPReadinessProve(portName, path, scheme. ..`?

**Review by matej-g — APPROVED** [review]
Thanks @clyang82! Looking good :+1:

**Pull request merged**